Elasticsearch/Document/DocumentCollection.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EMS\CommonBundle\Elasticsearch\Document;
+
+use EMS\CommonBundle\Elasticsearch\Response\ResponseInterface;
+
+final class DocumentCollection implements DocumentCollectionInterface
+{
+    /** @var array */
+    private $documents;
+
+    private function __construct()
+    {
+    }
+
+    public static function fromResponse(ResponseInterface $response): self
+    {
+        $collection = new static();
+
+        foreach ($response->getDocuments() as $document) {
+            $collection->add($document);
+        }
+
+        return $collection;
+    }
+
+    public function add(DocumentInterface $document): void
```

Comment:
should be private I think

Reply:
No I don't aggree, you should be able to create a collection yourself